### ISSUE TITLE
[FEATURE] Recommened config

### ISSUE THREAD

**Issue by hellcry37** [mmakaay/esphome-yeelight_bs2#61]
**Is your feature request related to a problem? Please describe.**
I am a noob when it comes to ESPHome and everything related to it, but i really wanted to get my Bedside Lamp 2 out of the cloud and I found this project. After the Xiaomi fiasco with turning off LAN control for Yeelight lights this project was born and I am fallowing it from the time dev offered some firmware on Yeelight forums.
To cut it short it was very hard for me to find a starting config that can do everything the lamp does by default. Example config is only a starting blank template with some options. So basically i found a conf on the internet that failed badly and corrected that step by step with update to v2021.10 doc.

After a lot of trial and error and help from two guys in esphome discord (one of them being the dev itself that did this awesome project) I got it working.

The learning curve was steep.

**Describe the solution you'd like**
I believe for people convenience it would be best to have a recommended config with all the things that lamp can do explained, something that will be added upon when necessary (new releases)

**Comment by mmakaay** [comment]
Okay, can you explain to me what you are missing exactly from the example.yaml configuration file?
As far as I know, this contains all the things needed for implementing the standard behavior that resembles the original behavior a lot. Furthermore, I have the configuration guide that explains what all the configuration options do and how they can be configured.

**Comment by mmakaay** [comment]
Also note that there is also some extra documentation to be found in the packages/*.yaml files that are included by the default.yaml.

**Comment by hellcry37** [comment]
Indeed, the docs are there, everything is docummented ok.

I was just referring to make this [example.yaml](https://github.com/mmakaay/esphome-xiaomi_bslamp2/blob/dev/example.yaml) to a [recommended.yaml](https://pastebin.com/w1WcL8jp) that contains all of them as per how you suggest lamp config, that's it, nothing more.

**Comment by mmakaay** [comment]
But that is what example.yaml does. It is the standard name used in many ESPHome projects for providing the standard / recommended / you-name-it configuration.
I don't see the point in copying the current example.yaml to recommended.yaml, ending up with two identical files to maintain.

**Comment by mmakaay** [comment]
Btw, the recommended.yaml that you link to is simply outdated and not one of mine. Not sure why you linked that one. 
It does look like it was based on the previous version of my example.yaml.

**Comment by hellcry37** [comment]
Like I said I'am new here, so you telling me this is all is needed in my config:
```yaml
# --------------------------------------------------------------------------
# Substitutions
#
# These are substitutions as used by the configuration packages from below.
# You can uncomment and update the ones that you want to modify.
# --------------------------------------------------------------------------

substitutions:
  #name: bedside-lamp
  #friendly_name: Bedside Lamp
  #light_name: ${friendly_name} RGBWW Light
  #light_mode_text_sensor_name: ${friendly_name} Light Mode
  #default_transition_length: 800ms

# --------------------------------------------------------------------------
# Load configuration packages
#
# These provide a convenient way to compose your device configuration from
# some functional building blocks. Pick and mix the blocks that you need.
#
# For customization you can override options in your config or you can
# copy the contents of these packages directly in your config file as
# an example for your own customizations.
#
# Available packages are:
# - core.yaml                : core components & hardware setup
# - behavior_default.yaml    : default device behavior
# - ota_feedback.yaml        : enable visual feedback during OTA updates
# - activate_preset_svc.yaml : 'activate_preset' service for Home Assistant
# --------------------------------------------------------------------------

packages:
  bslamp2:
    url: https://github.com/mmakaay/esphome-xiaomi_bslamp2
    ref: dev
    files:
      - packages/core.yaml
      - packages/behavior_default.yaml
      - packages/ota_feedback.yaml
      - packages/activate_preset_svc.yaml
    refresh: 0s

# --------------------------------------------------------------------------
# Use your own preferences for these components.
# --------------------------------------------------------------------------

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password

api:
  password: !secret api_password

ota:
  password: !secret ota_password
```

Because this is what I see in the example.yaml

The rest is imported from packages?

**Comment by hellcry37** [comment]
OK I understand now, that is all that is needed, the rest is referenced in packages. Ty i consider this matter close and sry for any inconvenience.

**Comment by mmakaay** [comment]
Ah, I get your confusion now. Yes, that is indeed all that is needed for a basic setup 😄 

The latest ESPHome allows for config packages that are stored on GitHub, so I leveraged that to make the example.yaml as small as possible. Those big configs are either a blast from the past, or the result of wanting to customize a lot of things. In case of the latter, you'd remove a yaml package from this config and copy its contents from packages/the.yaml into your main yaml file. Only then your yaml file would grow.

Great we got this sorted out 👍
